tests/unit/src/unit/issues/Issue11807.hx

```diff
@@ -0,0 +1,13 @@
+package unit.issues;
+
+class Issue11807 extends Test {
+	function test() {
+		var test = null;
+
+		do {
+			test = {type: 5};
+		} while (test.type != 5);
+
+		utest.Assert.pass();
```

Comment:
It would be better to have `eq(5, test.type)` here. Otherwise the test would pass even if the loop is completely ignored.

Reply:
Its meant to be to see if it crashes or not, but sure